lib/Network/OAuth2/Server/API.hs

```diff
@@ -566,12 +566,28 @@ serverPostToken
     -> m Html
 -- | Revoke a given token
 serverPostToken ref user_id _ (DeleteRequest token_id) = do
-    success <- liftIO $ storeRevokeToken ref user_id token_id
-    if success then
-        let link = safeLink (Proxy :: Proxy AnchorOAuth2API) (Proxy :: Proxy ListTokens) page1
-        in throwError err302{errHeaders = [(hLocation, B.pack $ show link)]} --Redirect to tokens page
-    else
-        throwError err500{errBody = "Something quite horrible has happened"}
+    maybe_tok <- liftIO $ storeReadToken ref (Right token_id)
+    tok <- case maybe_tok of
+        Nothing -> throwError err403 { errBody = "Invalid token_id" }
+        Just (_, tok) -> return tok
+    case tokenDetailsUserID tok of
+        Nothing -> do
+            liftIO . errorM logName $
+                "user_id " <> show user_id <> " tried to revoke token_id " <>
+                show token_id <> ", which did not have a user_id"
+            throwError err403 { errBody = "Invalid token_id" }
```

Comment:
It should be exactly the same as any other error, so the client doesn't get extra information it shouldn't. Hence they are all the same. A 404 surely not, I would have said 400 maybe.